### PR TITLE
Add ha-sortable component

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Introduce `ha-sortable` to easily add sorting feature to lists instead of manually creating sortable instance. The component itself doesn't use shadowdom rendering to support sortable-js.

The sidebar had a different logic because it was older code. I updated to the new sortable component and it's waaaay cleaner!

`ha-sortable-style` has been renamed to `ha-sidebar-edit-style` because sortable related style has been moved to `ha-sortable` component. We no longer include these styles in other components 🎉.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-sortable.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+export class HaSortable extends LitElement {
+  private _sortable?: SortableInstance;
+
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ type: String }) public container?: string;
+
+  @property({ type: String }) public item?: string;
+
+  @property({ type: String }) public handle?: string;
```
Comment (bramkragten) [inline src/components/ha-sortable.ts:27]: I would make it cleared that this is a selector, like name it `itemSelector`, `containerSelector`, `handleSelector`, or make it accept both a selector string and an element

Comment (piitaya) [reply src/components/ha-sortable.ts:27]: I will rename it to selector 🙂

Comment (bramkragten) [reply src/components/ha-sortable.ts]: container could be an element I guess, but selectors are probably easier

Comment (piitaya) [reply src/components/ha-sortable.ts]: It's not used and I don't think it will be used. Picking the first child is ok.

**Comment by silamon** [comment]
sortable.core.esm.js:1867 Uncaught TypeError: Cannot read properties of null (reading 'parentNode')
    at Sortable._onDragOver (sortable.core.esm.js:1867:29)
    at Sortable.handleEvent (sortable.core.esm.js:2062:16)

I'm getting the error above in the choose action reorder mode when I attempt to order any action in a list of 3 options. You can only see it if you open the browser console.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**